runtime/src/tests/locks.rs

```diff
@@ -64,3 +64,48 @@ fn incompatible_stakes_check_passed_successfully() {
         );
     });
 }
+
+#[test]
+fn incompatible_stakes_for_staking_pallet() {
```

Comment:
Would it be possible to test the SignedExtension behavior in this test? (ie. that it actually blocks the transaction?)

Reply:
It should be possible to write a test that instantiates the extension, and we invoke pre_dispatch() and validate(), but for a more realistic test we would have to maybe do it in the node package, instantiate a transaction pool and test by injecting extrinsics into it. It might make sense to add, but I went for integration test instead. WDYT